### ISSUE TITLE
The script tr_drc_to_klayout_drc.py causing memory issues.

### ISSUE THREAD

**Issue by jurevreca12** [The-OpenROAD-Project/OpenLane#1108]
### Prompt
The script `tr_drc_to_klayout_drc.py` used to convert the results to an klayout xml style file is consuming large amounts of memory due to the way it is written. Concretely what happened to me is that magic found 11149122 DRC issues with my design, and this lead the script `tr_drc_to_klayout_drc.py` to consume too much memory and crash. 

### Proposal
I did not study this script in detail, but it seems that the situation is similar to this:
https://stackoverflow.com/questions/31963124/memory-leakish-when-using-re-and-mmap
Basically instead of parsing the whole file at once, you parse in chunks. This should minimize memory consumption,
and resolve this issue.

EDIT: grammar fix
